### PR TITLE
fix(cdk): correct selection from textfield in firefox

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

### CODE REVIEW

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/cdk/utils/dom/get-selection-text.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * @description:
+ * cross browser way to get selected text
+ *
+ * History:
+ * BUG - window.getSelection() fails when text selected in a form field
+ * https://bugzilla.mozilla.org/show_bug.cgi?id=85686
+ */
+export function tuiGetSelectionText(windowRef: Window, target?: Target): string | null {
```
Comment (waterplea) [inline projects/cdk/utils/dom/get-selection-text.ts:11]: What about copying not from an input field? There's got to be target in that event as well.

File: projects/cdk/utils/dom/get-selection-text.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    const selection = windowRef.getSelection();
```
Comment (waterplea) [inline projects/cdk/utils/dom/get-selection-text.ts:12]: How about we pass `document` instead so we only have one argument?
https://developer.mozilla.org/en-US/docs/Web/API/Document/getSelection
We can then use `activeElement` to test if we are in an input field:
```js
document.activeElement.value.substring(
  document.activeElement.selectionStart,
  document.activeElement.selectionEnd
)
```

Comment (splincode) [reply projects/cdk/utils/dom/get-selection-text.ts]: fixed

File: projects/cdk/utils/dom/get-selection-text.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/utils/dom/get-selection-text.ts]: fixed

File: projects/cdk/utils/dom/get-selected-text.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * History:
+ * BUG - window.getSelection() fails when text selected in a form field
+ * https://bugzilla.mozilla.org/show_bug.cgi?id=85686
+ */
+export function tuiGetSelectedText(windowRef: Window): string | null {
+    const selection = windowRef.getSelection();
+    const element = windowRef.document.activeElement as
+        | HTMLInputElement
+        | HTMLTextAreaElement
```
Comment (nsbarsukov) [inline projects/cdk/utils/dom/get-selected-text.ts:13]: let use type guard
```ts
function isFormFieldElement(
    element: Target,
): element is HTMLInputElement | HTMLTextAreaElement {
    return (
        !!element &&
        'value' in element &&
        'selectionStart' in element &&
        'selectionEnd' in element
    );
}
```

Comment (splincode) [reply projects/cdk/utils/dom/get-selected-text.ts]: fixed

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**